GenericTest.py

```diff
@@ -214,38 +214,59 @@ def prepare_CORS(self, method):
         headers['Access-Control-Request-Headers'] = "Content-Type"  # Needed for POST/PATCH etc only
         return headers
 
-    def validate_CORS(self, method, response):
+    # 'check' functions return a Boolean pass/fail indicator and a message
+    def check_CORS(self, method, headers):
         """Check the CORS headers returned by an API call"""
-        if 'Access-Control-Allow-Origin' not in response.headers:
-            return False
+        if 'Access-Control-Allow-Origin' not in headers:
+            return False, "Incorrect CORS headers: {}".format(headers)
         if method == "OPTIONS":
-            if 'Access-Control-Allow-Headers' not in response.headers:
-                return False
-            if 'Access-Control-Allow-Methods' not in response.headers:
-                return False
-            if method not in response.headers['Access-Control-Allow-Methods']:
-                return False
-        return True
+            if 'Access-Control-Allow-Headers' not in headers:
+                return False, "Incorrect CORS headers: {}".format(headers)
+            if 'Access-Control-Allow-Methods' not in headers:
+                return False, "Incorrect CORS headers: {}".format(headers)
+            if method not in headers['Access-Control-Allow-Methods']:
+                return False, "Incorrect CORS headers: {}".format(headers)
+        return True, ""
+
+    def check_content_type(self, headers):
+        """Check the Content-Type header of an API request or response"""
+        if "Content-Type" not in headers:
+            return False, "API failed to signal a Content-Type."
+        else:
+            ctype = headers["Content-Type"]
+            ctype_params = ctype.split(";")
+            if ctype_params[0] != "application/json":
+                return False, "API signalled a Content-Type of {} rather than application/json." \
+                              .format(ctype)
+            elif len(ctype_params) == 2 and ctype_params[1].strip() == "charset=utf-8":
```

Comment:
Looks like this may need the same fix as https://github.com/AMWA-TV/nmos-testing/pull/346/files